### ISSUE TITLE
Build WireGuard for macOS/iOS problems

### ISSUE THREAD

**Issue by notecode** [zakosaba/wireguard-macos-app#6]
I'v just built WireGuard client successfully with the official source code. With Xcode 14.2 on macOS12.6.
But, when running the app on macOS, I can not activate a tunnel. I got the error alert:

"Activation failure. The tunnel could not be activated. Please ensure that you are connected to the Internet."

![wg-error](https://github.com/user-attachments/assets/0fe70553-b66e-4867-ab19-cd38ea9314e2)

I can not get help from the official, so come to get help from you.  I have used your release, which can be used successfully on my mac, so you must have solved this problem.  I am not good at obj-c / swift programming, it hard for me to solve it.  I think it maybe some network entitlements configuration needed. 

![wg-error2](https://github.com/user-attachments/assets/96c4552f-166b-436b-8637-7c8a88375786)

Could you give me some advice? Thank you.

**Comment by vaardan** [comment]
Hi, honestly I have no idea. I've tried to compile it once, it didn't work (like at all), I gave up and just downloaded application from AppStore and then re-uploaded it here. 

There can be many issues, mismatched versions of some stuff, missing components or libraries, etc. The error you got isn't helpful, so you might have to take a look at the actual logs for more info, wherever they may be. If the official devs are not responding, your best shot would be to find the informative error in the logs and search it on the Internet.

Regardless, IMHO just used the stuff I uploaded, if you say it works for you. Because even if you manage to make compiled version work, as far as I know you will still have issues with app not being properly signed and stuff.
